valve/server_query.go

```diff
@@ -120,6 +120,24 @@ func (this *ServerQuerier) a2s_info(info *ServerInfo) error {
 		return err
 	}
 
+	switch data[4] {
```

Comment:
length needs to be checked here.

Reply:
The convention in this code is that you don't need to do these checks. Callers should use the Try() helper which can catch exceptions.